src/main/java/frc/robot/subsystems/drivetrain/BaseSwerveSubsystem.java

```diff
@@ -173,10 +179,11 @@ public void periodic() {
             if(ledSubsystem != null){
                 ledSubsystem.tagDetected();
             }
-            poseEstimator.addVisionMeasurement(
+            if(poseInField(visionPose.estimatedPose.toPose2d())){
```

Comment:
super nitpicky but it might make sense to do a if !poseInField return; instead, only because it'll be easier to add other filters later on.